bin/setup

```diff
@@ -30,47 +22,56 @@ end.parse!
 # path to your application root.
 APP_ROOT = Pathname.new File.expand_path('../../', __FILE__)
 
-Dir.chdir APP_ROOT do
-  puts "== Installing dependencies =="
+def system!(*args)
+  system(*args) || abort("\n== Command #{args} failed ==")
+end
+
+chdir APP_ROOT do
```

Comment:
Looks like this commit explains it (was looking at the `bin/update` history previously):

https://github.com/rails/rails/commit/1ae88b84a6142e945f989c82a6b0ffbeac4a8173

That said, we are not really following that too well as we are just making system calls ourselves to `cp`.